### ISSUE TITLE
feat: `shorebird patch` should support patching multiple platforms at once

### ISSUE THREAD

**Issue by felangel** [shorebirdtech/shorebird#1762]
**Description**

As a user, I want to be able to patch multiple platforms (e.g iOS and Android) at once and have a single patch number associated with both since the patch contains identical changes.

Currently, even if I want to patch the exact same code for iOS and Android, I have to run `shorebird patch android` and `shorebird patch ios` which results in patch 1 and patch 2 respectively and I have to remember that patch 1 and patch 2 contain identical code.

One way we could solve this is by restructuring `shorebird patch` to be a single command with `platforms` as an option:

```sh
# This will patch all platforms for the current release
shorebird patch

# This will only patch iOS
shorebird patch --platforms=ios

# This will only patch Android
shorebird patch --platforms=android

# This will only patch iOS and MacOS (once we support support desktop)
shorebird patch --platforms=ios,macos
```

With the proposed solution, running `shorebird patch` would be reflected as a single patch with multiple platforms on the console and `package:shorebird_code_push` would return `1` as the current patch number on all platforms which would eliminate any confusion/book-keeping needed when applying platform-agnostic fixes.

**Considerations**

With the proposed solution, we'd likely want to build all patches first (and if any fail, abort with a summary) and only if all patches are built successfully would be upload them (with a retry for upload failures).

We'd also potentially want to adjust the console to support deactivating a patch for a specific set of platforms.

**Comment by wolfenrain** [comment]
This would make it so much easier for me as a user to identify which patch is code-wise shared with other platforms! And the benefit of only being able to patch when all platform patches succeed in building is a nice addition!

**Comment by eseidel** [comment]
This is now done, and will be released shortly as part of version 1.1.  This was a side effect of https://github.com/shorebirdtech/shorebird/issues/1988.

**Comment by erickzanardo** [comment]
Extracted the "same patch number across platforms" piece of this issue to a standalone one: https://github.com/shorebirdtech/shorebird/issues/2068
